### PR TITLE
Add a litmusbook for prometheus deployment

### PR DESCRIPTION
**Pull request by shashank855**
**What this PR does / why we need it**:
- Add a litmus book for **Prometheus Deployment**.
**Following the the log of the litmus job:**
```
PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [127.0.0.1]

TASK [Record test instance/run ID] *********************************************
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
changed: [127.0.0.1] => {"changed": true, "checksum": "8d34e98f66235fb9c3570a618810bfbb97f90cca", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "279779df6782cc15347dd36901dad868", "mode": "0644", "owner": "root", "size": 422, "src": "/root/.ansible/tmp/ansible-tmp-1539092233.58-173394792107143/source", "state": "file", "uid": 0}

TASK [Apply the litmus result CR] **********************************************
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:01.215395", "end": "2018-10-09 13:37:16.588127", "failed_when_result": false, "rc": 0, "start": "2018-10-09 13:37:15.372732", "stderr": "", "stderr_lines": [], "stdout": "litmusresult \"prometheus-deployment\" configured", "stdout_lines": ["litmusresult \"prometheus-deployment\" configured"]}

TASK [Check whether the provider storageclass is applied] **********************
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get storageclass openebs-standard", "delta": "0:00:00.857121", "end": "2018-10-09 13:37:17.876457", "rc": 0, "start": "2018-10-09 13:37:17.019336", "stderr": "", "stderr_lines": [], "stdout": "NAME               PROVISIONER\nopenebs-standard   openebs.io/provisioner-iscsi", "stdout_lines": ["NAME               PROVISIONER", "openebs-standard   openebs.io/provisioner-iscsi"]}

TASK [Replace the pvc placeholder with provider] *******************************
changed: [127.0.0.1] => {"changed": true, "msg": "2 replacements made"}

TASK [Replace the storageclass placeholder with provider] **********************
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Get the application label values from env] *******************************
ok: [127.0.0.1] => {"ansible_facts": {"app_lkey": "name", "app_lvalue": "prometheus"}, "changed": false}

TASK [Replace the application label placeholder] *******************************
changed: [127.0.0.1] => {"changed": true, "msg": "4 replacements made"}

TASK [Create test specific namespace.] *****************************************
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Checking the status  of test specific namespace.] ************************
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get namespace litmus -o custom-columns=:status.phase --no-headers", "delta": "0:00:00.830972", "end": "2018-10-09 13:37:20.952178", "rc": 0, "start": "2018-10-09 13:37:20.121206", "stderr": "", "stderr_lines": [], "stdout": "Active", "stdout_lines": ["Active"]}

TASK [Deploying prometheus] ****************************************************
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f prometheus.yml -n litmus", "delta": "0:00:01.088637", "end": "2018-10-09 13:37:22.473626", "rc": 0, "start": "2018-10-09 13:37:21.384989", "stderr": "", "stderr_lines": [], "stdout": "deployment \"prometheus\" created\npersistentvolumeclaim \"prometheus-claim\" created\nservice \"prometheus\" created", "stdout_lines": ["deployment \"prometheus\" created", "persistentvolumeclaim \"prometheus-claim\" created", "service \"prometheus\" created"]}

TASK [Checking prometheus is in running state] *********************************
FAILED - RETRYING: Checking prometheus is in running state (100 retries left).
FAILED - RETRYING: Checking prometheus is in running state (99 retries left).
FAILED - RETRYING: Checking prometheus is in running state (98 retries left).
FAILED - RETRYING: Checking prometheus is in running state (97 retries left).
FAILED - RETRYING: Checking prometheus is in running state (96 retries left).
changed: [127.0.0.1] => {"attempts": 6, "changed": true, "cmd": "kubectl get pods -n litmus -o jsonpath='{.items[?(@.metadata.labels.name==\"prometheus\")].status.phase}'", "delta": "0:00:00.954326", "end": "2018-10-09 13:38:19.951066", "rc": 0, "start": "2018-10-09 13:38:18.996740", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

TASK [set_fact] ****************************************************************
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
changed: [127.0.0.1] => {"changed": true, "checksum": "ed42cf3bf2b6454e836fc742f64dd77c7bcf4861", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "915c620161e6dace4ecd5d2ec94d5e8d", "mode": "0644", "owner": "root", "size": 420, "src": "/root/.ansible/tmp/ansible-tmp-1539092300.27-95936637097671/source", "state": "file", "uid": 0}

TASK [Apply the litmus result CR] **********************************************
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:01.020678", "end": "2018-10-09 13:38:23.315804", "failed_when_result": false, "rc": 0, "start": "2018-10-09 13:38:22.295126", "stderr": "", "stderr_lines": [], "stdout": "litmusresult \"prometheus-deployment\" configured", "stdout_lines": ["litmusresult \"prometheus-deployment\" configured"]}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=14   changed=11   unreachable=0    failed=0   

```

### CODE REVIEW

File: apps/prometheus/deployers/run_litmus_test.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      name: litmus
+      labels:
+        app: prometheus-deployment-litmus
+
+    spec:
+      serviceAccountName: litmus
+      restartPolicy: Never
+      containers:
+      - name: ansibletest
+        image: ranjnshashank855/ansible-runner:test
```
Comment (nsathyaseelan) [inline apps/prometheus/deployers/run_litmus_test.yml]: Change the image name

File: apps/prometheus/deployers/test.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+        - name: Deploying prometheus
+          shell: kubectl apply -f {{ prometheus_deployment }} -n {{ app_ns }}
+
+        - name: Checking prometheus is in running state
+          shell: kubectl get pods -n {{ app_ns }} -o jsonpath='{.items[?(@.metadata.labels.{{app_lkey}}=="{{app_lvalue}}")].status.phase}'
+          register: result
+          until: "'Running' in result.stdout"
+          delay: 10
+          retries: 100
```
Comment (nsathyaseelan) [inline apps/prometheus/deployers/test.yml]: Can you reduce the retries

Comment (dargasudarshan) [reply apps/prometheus/deployers/test.yml]: @shashank855 30*10 should be good enough..

File: apps/prometheus/deployers/test.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+      always:
+            ## RECORD END-OF-TEST IN LITMUS RESULT CR
+        - name: Generate the litmus result CR to reflect EOT (End of Test)
+          template:
+            src: /litmus-result.j2
+            dest: litmus-result.yaml
+          vars:
+            test: "{{ test_name }}"
+            app: ""
```
Comment (gprasath) [inline apps/prometheus/deployers/test.yml]: Update the application name to LR

File: apps/prometheus/deployers/test.yml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          when: lookup('env','RUN_ID')
+
+         ## RECORD START-OF-TEST IN LITMUS RESULT CR
+        - name: Generate the litmus result CR to reflect SOT (Start of Test)
+          template:
+            src: /litmus-result.j2
+            dest: litmus-result.yaml
+          vars:
+            test: "{{ test_name }}"
+            app: ""
```
Comment (gprasath) [inline apps/prometheus/deployers/test.yml]: Update the application name to LR

**Pull request merged**